### ISSUE TITLE
getting `shift on reference is experimental` on empty diffs

### ISSUE THREAD

**Issue by mmlb** [so-fancy/diff-so-fancy#114]
```
~/cloned/so-fancy/diff-so-fancy master
❯ git status
On branch master
Your branch is up-to-date with 'origin/master'.
nothing to commit, working directory clean

~/cloned/so-fancy/diff-so-fancy master
❯ git diff
shift on reference is experimental at /usr/share/diff-so-fancy/header_clean/header_clean.pl line 113.

~/cloned/so-fancy/diff-so-fancy master
❯ git diff --cached
shift on reference is experimental at /usr/share/diff-so-fancy/header_clean/header_clean.pl line 113.

~/cloned/so-fancy/diff-so-fancy master
❯ perl --version

This is perl 5, version 22, subversion 1 (v5.22.1) built for x86_64-linux-thread-multi

Copyright 1987-2015, Larry Wall

Perl may be copied only under the terms of either the Artistic License or the
GNU General Public License, which may be found in the Perl 5 source kit.

Complete documentation for Perl, including FAQ lists, should be found on
this system using "man perl" or "perldoc perl".  If you have access to the
Internet, point your browser at http://www.perl.org/, the Perl Home Page.

```

**Comment by xerebz** [comment]
Seeing this when trying any git diff using the same perl version as @mmlb

**Comment by mmlb** [comment]
It works for me if there is some content in the underlying diff.

**Comment by jeffcasavant** [comment]
Happens to me, suspect this is related to the perl version.  I have the same as mmlb.

From https://rt.cpan.org/Public/Bug/Display.html?id=101680:
> There's a new warning in newer perl versions (5.20.x, 5.21.x): "shift on reference is experimental". See http://www.cpantesters.org/cpan/report/203b8f80-a22a-11e4-931f-81ede0bfc7aa for a sample test report where this happens.

**Comment by paulirish** [comment]
cc @scottchiefbaker if you have ideas

**Comment by scottchiefbaker** [comment]
@mmlb Ahhhhh this should be an easy fix. Try changing line 113 to 

```
shift(@$foo);
```

If that fixes the problem for you, I'll send a PR.

**Comment by paulirish** [comment]
I just got a repro of that bug going..


```sh
brew install --force-bottle perl
export PERL5LIB="/Users/paulirish/.homebrew/Cellar/perl/5.22.1/lib/5.22.1/"
# break the pipe into header_clean.pl 
git diff | ./diff-so-fancy | /Users/paulirish/.homebrew/Cellar/perl/5.22.1/bin/perl libs/header_clean/header_clean.pl
```

![image](https://cloud.githubusercontent.com/assets/39191/13614728/c742abd4-e526-11e5-9665-39f836a26b72.png)


Just updated the source with @scottchiefbaker's fix.. and we're good!

**Comment by scottchiefbaker** [comment]
My dev environment has Perl 5.18, and this was introduced in 5.20+

My bad...

**Comment by paulirish** [comment]
We've merged this to master. 

If you dont want to wait for a new release on npm, you can clone from master and do a `npm link`

**Comment by vanniktech** [comment]
I also had this issue on my Ubuntu 15.10 though I don't know the perl version that's running on it right now.

**Comment by mmlb** [comment]
:+1: thanks! :tada:

**Comment by paulirish** [comment]
This is now shipped in 0.6.1+
